File: padroesComportamentais/padrao-de-projeto-observer/java/src/main/java/observer/Produto.java
package observer;

public class Produto {
    private String nome;
    private Float preco;
    private Float promocao;

    public String getNome() {
        return nome;
    }

    public Produto setNome(String nome) {
        this.nome = nome;
        return this;
    }

    public Float getPreco() {
        return this.preco - this.preco * (this.getPromocao() / 100);
    }

    public Produto setPreco(Float preco) {
        this.preco = preco;
        return this;
    }

    public Produto setPromocao(float desconto) {
        this.promocao = desconto;
        return this;
    }

    public Float getPromocao() {
        return this.promocao;
    }

    @Override
    public String toString() {
        return "nome do produto: " + this.getNome() + " | preco: " + this.getPreco();
    }
}

File: padroesComportamentais/padrao-de-projeto-observer/php/src/Produto.php
<?php

namespace Observer\Php;

class Produto
{
    private string $nome;
    private float $preco;
    private float $promocao;

    public function getNome(): string
    {
        return $this->nome;
    }

    public function setNome(string $nome): Produto
    {
        $this->nome = $nome;
        return $this;
    }

    public function getPreco(): float
    {
        return $this->preco - $this->preco * $this->promocao / 100;
    }

    public function setPreco(float $preco): Produto
    {
        $this->preco = $preco;
        return $this;
    }

    public function getPromocao(): float
    {
        return $this->promocao;
    }

    public function setPromocao(float $promocao): Produto
    {
        $this->promocao = $promocao;
        return $this;
    }
}